frontend/src/components/main-page/groups-tree-page/group-details/group-details.tsx

```diff
@@ -14,7 +28,72 @@ export const GroupDetails: React.FC = () => {
 
   return (
     <React.Fragment>
-      Group Details {JSON.stringify(controller.group)}
+      {controller.group && (
+        <Box
+          sx={{
+            overflowX: 'hidden',
+            overflowY: 'auto',
+            padding: 4,
+            maxWidth: '700px',
+          }}
+        >
+          <Typography variant="h3">Group Information</Typography>
+
+          {Object.keys(controller.group).includes('name') && (
+            <List>
+              <ListItem divider>
+                <ListItemIcon>
+                  <Badge />
+                </ListItemIcon>
+                <ListItemText
+                  primary={(controller.group as ServiceDocsGroup).name}
+                  secondary="Name"
+                />
+              </ListItem>
+            </List>
+          )}
+
+          {Object.keys(controller.group).includes('identifier') && (
+            <List>
+              <ListItem divider>
+                <ListItemIcon>
+                  <Group />
+                </ListItemIcon>
+                <ListItemText
+                  primary={(controller.group as ServiceDocsGroup).identifier}
+                  secondary="Full identifier"
+                />
+              </ListItem>
+            </List>
+          )}
+
+          <List>
+            <ListItem divider>
+              <ListItemIcon sx={{ color: 'inherit' }}>
+                <CenterFocusStrongOutlined />
+              </ListItemIcon>
+              <ListItemText
+                primary={`${controller.group.services.length} Services`}
+                secondary="Number of owned services"
+              />
+            </ListItem>
+          </List>
+
+          <List>
+            <ListItem divider>
+              <ListItemIcon sx={{ color: 'inherit' }}>
+                <DatasetOutlined />
+              </ListItemIcon>
+              <ListItemText
+                primary={`${
+                  Object.keys(controller.group.childGroups).length
+                } Groups`}
+                secondary="Number of owned groups"
+              />
+            </ListItem>
+          </List>
```

Comment:
Just a small suggestion to show "1 Service" instead of "1 Services" (and the same for groups):

```suggestion
          <List>
            <ListItem divider>
              <ListItemIcon sx={{ color: 'inherit' }}>
                <CenterFocusStrongOutlined />
              </ListItemIcon>
              <ListItemText
                primary={`${controller.group.services.length} ${
                  controller.group.services.length === 1
                    ? 'Service'
                    : 'Services'
                }`}
                secondary="Number of owned services"
              />
            </ListItem>
          </List>

          <List>
            <ListItem divider>
              <ListItemIcon sx={{ color: 'inherit' }}>
                <DatasetOutlined />
              </ListItemIcon>
              <ListItemText
                primary={`${
                  Object.keys(controller.group.childGroups).length
                } ${
                  Object.keys(controller.group.childGroups)
                    .length === 1
                    ? 'Group'
                    : 'Groups'
                }`}
                secondary="Number of owned groups"
              />
            </ListItem>
          </List>
```

If you are using the "groupsWithType" approach from the other comment of mine, then this would be the correct snippet:
```suggestion
          <List>
            <ListItem divider>
              <ListItemIcon sx={{ color: 'inherit' }}>
                <CenterFocusStrongOutlined />
              </ListItemIcon>
              <ListItemText
                primary={`${controller.groupWithType.group.services.length} ${
                  controller.groupWithType.group.services.length === 1
                    ? 'Service'
                    : 'Services'
                }`}
                secondary="Number of owned services"
              />
            </ListItem>
          </List>

          <List>
            <ListItem divider>
              <ListItemIcon sx={{ color: 'inherit' }}>
                <DatasetOutlined />
              </ListItemIcon>
              <ListItemText
                primary={`${
                  Object.keys(controller.groupWithType.group.childGroups).length
                } ${
                  Object.keys(controller.groupWithType.group.childGroups)
                    .length === 1
                    ? 'Group'
                    : 'Groups'
                }`}
                secondary="Number of owned groups"
              />
            </ListItem>
          </List>
```